File: C++/2997.cpp
#include <vector>

class Solution {
public:
    int minOperations(std::vector<int>& nums, int k){
        int xor_val = 0, ans = 0;

        for(int i: nums) xor_val ^= i;
        
        while(k || xor_val){
            ans += ((k%2) != (xor_val%2));
            k /= 2;
            xor_val /= 2;
        }
        return ans;
    }
};

// Topic: Bit Manipulation
// Time Complexity: O(n)
// Space Complexity: O(1)

File: Python/2997.py
from typing import List

class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        xor, ans = 0, 0

        for i in nums: xor ^= i

        while k or xor:
            ans += ((k%2)!= (xor%2))
            k //= 2
            xor //= 2

        return ans

# Topic: Bit Manipulation
# Time Complexity: O(n)
# Space Complexity: O(1)        
        